src/elements/content-uploader/UploadState.tsx

```diff
@@ -0,0 +1,114 @@
+    onSelect,
+    isFolderUploadEnabled,
+}: UploadStateProps) => {
+    const intl = useIntl();
+    let icon;
+    let content;
+    switch (view) {
+        case VIEW_ERROR:
+            icon = <HatWand aria-label={intl.formatMessage(messages.uploadErrorState)} height={126} width={130} />;
+            content = <UploadStateContent message={<FormattedMessage {...messages.uploadError} />} />;
+            break;
+        case VIEW_UPLOAD_EMPTY:
+            icon = <UploadEmptyState title={<FormattedMessage {...messages.uploadEmptyState} />} />;
+            /* eslint-disable no-nested-ternary */
+            content =
+                canDrop && hasItems ? (
+                    <UploadStateContent message={<FormattedMessage {...messages.uploadInProgress} />} />
+                ) : isTouch ? (
+                    <UploadStateContent
+                        fileInputLabel={<FormattedMessage {...messages.uploadNoDragDrop} />}
+                        onChange={onSelect}
+                        useButton
+                    />
+                ) : (
+                    <UploadStateContent
+                        fileInputLabel={<FormattedMessage {...messages.uploadEmptyFileInput} />}
+                        folderInputLabel={
+                            isFolderUploadEnabled && <FormattedMessage {...messages.uploadEmptyFolderInput} />
+                        }
+                        message={
+                            isFolderUploadEnabled ? (
+                                <FormattedMessage {...messages.uploadEmptyWithFolderUploadEnabled} />
+                            ) : (
+                                <FormattedMessage {...messages.uploadEmptyWithFolderUploadDisabled} />
+                            )
+                        }
+                        onChange={onSelect}
+                    />
+                );
+            /* eslint-enable no-nested-ternary */
+            break;
+        case VIEW_UPLOAD_IN_PROGRESS:
+            icon = <UploadEmptyState title={<FormattedMessage {...messages.uploadEmptyState} />} />;
+            content = <UploadStateContent message={<FormattedMessage {...messages.uploadInProgress} />} />;
+            break;
+        case VIEW_UPLOAD_SUCCESS:
+            icon = <UploadSuccessState title={<FormattedMessage {...messages.uploadSuccessState} />} />;
+            content = (
+                <UploadStateContent
+                    fileInputLabel={<FormattedMessage {...messages.uploadSuccessFileInput} />}
+                    folderInputLabel={
+                        isFolderUploadEnabled && <FormattedMessage {...messages.uploadSuccessFolderInput} />
+                    }
+                    message={<FormattedMessage {...messages.uploadSuccess} />}
+                    onChange={onSelect}
+                    useButton={isTouch}
+                />
+            );
+            break;
+        default:
+            break;
+        /* eslint-enable jsx-a11y/label-has-for */
```

Comment:
i think this eslint can be removed. it's a deprecated rule and i don't see `label`s used in this file